clash-lib/src/Clash/Normalize.hs

```diff
@@ -117,6 +117,7 @@ runNormalization opts supply globals typeTrans reprs tcm tupTcm eval primMap rcs
     rwEnv     = RewriteEnv
                   (opt_dbgLevel opts)
                   (opt_dbgTransformations opts)
+                  (opt_aggressiveXOpt opts)
```

Comment:
Just a thought: maybe we should add the whole of `ClashOpts` to `RewriteEnv`, instead of adding each flag individually.

Reply:
A small subset of `ClashOpts` is added to `rwEnv` and a different small subset to `normState`, if `rwEnv` is available everywhere `normState` is, then I would say it's okay to just include the options in `rwEnv`.

But equally not every clash option is relevant to rewriting / normalization, so there's a case to be made for the current style too :man_shrugging:

Reply:
Yeah, it's not for now, that's for sure. I just dislike that we have to change that much for adding a simple flag.

Reply:
in this case it seems like just 2 extra lines though